backend/server/routers/courses.py

```diff
@@ -117,6 +130,14 @@ def search(string):
     code_query = coursesCOL.find({"code": {"$regex": pat}})
     title_query = coursesCOL.find({"title": {"$regex": pat}})
 
+    years = sorted(ARCHIVED_YEARS, reverse=True)
+    if len(list(code_query.clone())) == 0 and len(list(title_query.clone())) == 0:
+        for year in years:
+            code_query = archivesDB[str(year)].find({"code": {"$regex": pat}})
+            title_query = archivesDB[str(year)].find({"title": {"$regex": pat}})
+            if len(list(code_query.clone())) != 0 or len(list(title_query.clone())) != 0:
```

Comment:
is there a reason we have to clone here? does len(code_query) not work? Im worried that this may be an expensive computation as we expand our db.

Reply:
(another option can be a simple `if code_query or title_query:`

Reply:
oh ok i'll try that. i thought the default `false` behaviour only works for `None` value not on an empty iterator but will try to see if it works. Also to answer your question above, len(code_query) would consume the iterator and result in an empty list when we chain it so we need to clone() it.